src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -1809,12 +1816,19 @@ class RideConstructionWindow final : public Window
         widgets[WIDX_U_TRACK].type = WindowWidgetType::Empty;
         widgets[WIDX_O_TRACK].type = WindowWidgetType::Empty;
 
-        bool brakesSelected = _selectedTrackType == TrackElemType::Brakes
-            || _currentTrackCurve == (RideConstructionSpecialPieceSelected | TrackElemType::Brakes);
+        bool trackHasSpeedSetting = TrackTypeHasSpeedSetting(_selectedTrackType)
+            || TrackTypeHasSpeedSetting(_currentTrackCurve & ~RideConstructionSpecialPieceSelected);
         bool boosterTrackSelected = _selectedTrackType == TrackElemType::Booster
             || _currentTrackCurve == (RideConstructionSpecialPieceSelected | TrackElemType::Booster);
 
-        if (!brakesSelected && !boosterTrackSelected)
+        // Only necessary because TD6 writes speed and seat rotation to the same bits. Remove for new track design format.
+        bool trackHasSpeedAndSeatRotation = _selectedTrackType == TrackElemType::BlockBrakes
```

Comment:
This doesn’t actually check for seat rotation in any way, so why is it called this?

Reply:
Those are all the track elements that have a speed setting and support rotation. Technically it should be block brakes and any track element above HighestAlias since they cannot be written to td6.